src/tools/kdb/main.c

```diff
@@ -25,6 +60,44 @@ command subcommands[] = {
 	{ "get", addGetSpec, execGet },
 };
 
+cppCommand cppSubcommands[] = {
+	{ "basename", addBasenameSpec, execCppBasename },
+	{ "cp", addCpSpec, execCppCp },
+	{ "dirname", addDirnameSpec, execCppDirname },
+	{ "find", addFindSpec, execCppFind },
+	{ "ls", addLsSpec, execCppLs },
+	{ "merge", addMergeSpec, execCppMerge },
+	{ "meta-get", addMetaGetSpec, execCppMetaGet },
+	{ "meta-ls", addMetaLsSpec, execCppMetaLs },
+	{ "meta-rm", addMetaRmSpec, execCppMetaRm },
+	{ "meta-set", addMetaSetSpec, execCppMetaSet },
+	{ "meta-show", addMetaShowSpec, execCppMetaShow },
+	{ "mv", addMvSpec, execCppMv },
+	{ "namespace", addNamespaceSpec, execCppNamespace },
+	{ "rm", addRmSpec, execCppRm },
+	{ "set", addSetSpec, execCppSet },
+	{ "sget", addSgetSpec, execCppSget },
+	{ "export", addExportSpec, NULL },
+	{ "cache", addCacheSpec, execCppCache },
+	{ "complete", addCompleteSpec, execCppComplete },
+	{ "convert", addConvertSpec, execCppConvert },
+	{ "editor", addEditorSpec, execCppEditor },
+	{ "file", addFileSpec, execCppFile },
+	{ "gen", addGenSpec, execCppGen },
+	{ "import", addImportSpec, NULL },
+	{ "list-tools", addListToolsSpec, execCppListTools },
+	{ "mount", addMountSpec, NULL },
+	{ "plugin-check", addPluginCheckSpec, execCppPluginCheck },
+	{ "plugin-info", addPluginInfoSpec, execCppPluginInfo },
+	{ "plugin-list", addPluginListSpec, execCppPluginList },
+	{ "remount", addRemountSpec, execCppRemount },
+	{ "shell", addShellSpec, execCppShell },
+	{ "spec-mount", addSpecMountSpec, execCppSpecMount },
+	{ "test", addTestSpec, execCppTest },
+	{ "umount", addUmountSpec, execCppUmount },
+	{ "validate", addValidateSpec, execCppValidate },
+};
```

Comment:
I think using two arrays would make more sense to avoid iterating everything twice in case you reach line 267.

Something like:

```suggestion
cppCommand cppSubcommands[] = {
	{ "basename", addBasenameSpec, execCppBasename },
	{ "cp", addCpSpec, execCppCp },
	{ "dirname", addDirnameSpec, execCppDirname },
	{ "find", addFindSpec, execCppFind },
	{ "ls", addLsSpec, execCppLs },
	{ "merge", addMergeSpec, execCppMerge },
	{ "meta-get", addMetaGetSpec, execCppMetaGet },
	{ "meta-ls", addMetaLsSpec, execCppMetaLs },
	{ "meta-rm", addMetaRmSpec, execCppMetaRm },
	{ "meta-set", addMetaSetSpec, execCppMetaSet },
	{ "meta-show", addMetaShowSpec, execCppMetaShow },
	{ "mv", addMvSpec, execCppMv },
	{ "namespace", addNamespaceSpec, execCppNamespace },
	{ "rm", addRmSpec, execCppRm },
	{ "set", addSetSpec, execCppSet },
	{ "sget", addSgetSpec, execCppSget },
	{ "cache", addCacheSpec, execCppCache },
	{ "complete", addCompleteSpec, execCppComplete },
	{ "convert", addConvertSpec, execCppConvert },
	{ "editor", addEditorSpec, execCppEditor },
	{ "file", addFileSpec, execCppFile },
	{ "gen", addGenSpec, execCppGen },
	{ "list-tools", addListToolsSpec, execCppListTools },
	{ "plugin-check", addPluginCheckSpec, execCppPluginCheck },
	{ "plugin-info", addPluginInfoSpec, execCppPluginInfo },
	{ "plugin-list", addPluginListSpec, execCppPluginList },
	{ "remount", addRemountSpec, execCppRemount },
	{ "shell", addShellSpec, execCppShell },
	{ "spec-mount", addSpecMountSpec, execCppSpecMount },
	{ "test", addTestSpec, execCppTest },
	{ "umount", addUmountSpec, execCppUmount },
	{ "validate", addValidateSpec, execCppValidate },
};

cppCommand cppSubcommandsNoExec[] = {
	{ "export", addExportSpec, NULL },
	{ "import", addImportSpec, NULL },
	{ "mount", addMountSpec, NULL },
};
```

Then in the first `for` loop in line 190, you just iterate `cppSubcommandsNoExec` and only in line 267 you iterate `cppSubcommands`.